river/command/filter.zig

```diff
@@ -52,9 +49,82 @@ pub fn csdFilterAdd(
     args: []const []const u8,
     out: *?[]const u8,
 ) Error!void {
-    try appendFilter(
-        allocator,
-        &server.config.csd_filter,
-        args,
-    );
+    try modifyFilter(allocator, &server.config.csd_filter, args, .add);
+    csdFilterUpdateViews(args[1], .add);
+}
+
+pub fn csdFilterRemove(
+    allocator: *std.mem.Allocator,
+    seat: *Seat,
+    args: []const []const u8,
+    out: *?[]const u8,
+) Error!void {
+    try modifyFilter(allocator, &server.config.csd_filter, args, .remove);
+    csdFilterUpdateViews(args[1], .remove);
+}
+
+fn modifyFilter(
+    allocator: *std.mem.Allocator,
+    list: *std.ArrayList([]const u8),
+    args: []const []const u8,
+    operation: enum { add, remove },
+) Error!void {
```

Comment:
I think handling add and remove in the same function doesn't make much sense. The only reason I see for this choice is avoiding duplication of the length checking for `args`, which I don't think is worth it. That length checking is very simple and already duplicated a million places.